plugins/jvm-abi-gen/src/org/jetbrains/kotlin/jvm/abi/JvmAbiMetadataProcessor.kt

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright 2010-2020 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.jvm.abi
+
+import kotlinx.metadata.*
+import kotlinx.metadata.jvm.JvmClassExtensionVisitor
+import kotlinx.metadata.jvm.JvmPackageExtensionVisitor
+import kotlinx.metadata.jvm.KotlinClassHeader
+import kotlinx.metadata.jvm.KotlinClassMetadata
+import org.jetbrains.kotlin.load.java.JvmAnnotationNames.*
+import org.jetbrains.org.objectweb.asm.AnnotationVisitor
+import org.jetbrains.org.objectweb.asm.Opcodes
+
+/**
+ * Wrap the visitor for a Kotlin Metadata annotation to strip out private and local
+ * functions, properties, and type aliases as well as local delegated properties.
+ */
+fun abiMetadataProcessor(annotationVisitor: AnnotationVisitor): AnnotationVisitor =
+    kotlinClassHeaderVisitor { header ->
+        // kotlinx-metadata only supports writing Kotlin metadata of version >= 1.4, so we need to
```

Comment:
@sfs could you clarify a bit about scenario when this is necessery? I'm not familar enough with this tooling. Could it be used as separate tool? 
@udalov How good to force metadata upgrade in such way?

Reply:
I don't see any way around it. Supporting older metadata versions in `kotlinx-metadata-jvm` would be a bit complicated. Probably it should be explicitly documented in some kind of docs on `jvm-abi-gen` though

Reply:
Yeah, since this PR has started, we removed support of language/metadata version 1.3 from the compiler, so this is no longer a concern.